File: .net/dojos/dojo1/ThirdTry/ThirdTry/Frame.cs
namespace ThirdTry
{
    public class Frame
    {
        public int FirstBall { get; }
        public int SecondBall { get; }

        public Frame(int firstBall, int secondBall)
        {
            this.FirstBall = firstBall;
            this.SecondBall = secondBall;
        }

        public int Score => CalculateScore();
        public Frame Next { get; set; }

        private int CalculateScore()
        {
            return FirstBall+SecondBall+Bonus();
        }

        private int Bonus()
        {
            var isSpare = FirstBall != 10 && FirstBall + SecondBall == 10;
            var isStrike = FirstBall == 10;

            if (isSpare)
            {
                return Next.FirstBall;
            }
            if (isStrike)
            {
                return Next.FirstBall+Next.SecondBall;
            }
            return 0;
        }
    }
}

File: dotnet/dojos/dojo1/JBG/src/main/java/bowling/fourth/Frame.java
package bowling.fourth;

public class Frame {
    private final int firstRoll;
    private final int secondRoll;
    private Frame next;

    public Frame(int firstRoll, int secondRoll) {
        this.firstRoll = firstRoll;
        this.secondRoll = secondRoll;
    }

    public int countScore() {
        return ownScore() + countBonus();
    }

    private int countBonus() {
        int bonus = 0;
        if (isStrike()) {
            bonus = next.firstRoll + next.secondRoll;
        } else if (isSpare()) {
            bonus = next.firstRoll;
        }
        return bonus;
    }

    private int ownScore() {
        return firstRoll + secondRoll;
    }

    private boolean isStrike() {
        return firstRoll == 10;
    }

    private boolean isSpare() {
        return firstRoll + secondRoll == 10;
    }

    public void setNext(Frame next) {
        this.next = next;
    }
}
